bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -549,23 +546,32 @@ private void schedulePollingJob() {
         pollingJob = scheduler.scheduleWithFixedDelay(() -> {
             synchronized (sequenceLock) {
                 if (connector.isConnected()) {
-                    logger.debug("Polling the controller for updated status...");
-
-                    // poll each zone up to the number of zones specified in the configuration
-                    MonopriceAudioZone.VALID_ZONES.stream().limit(numZones).forEach((zoneName) -> {
+                    logger.debug("Polling the amplifier for updated status...");
+
+                    if (!disableKeypadPolling) {
+                        // poll each zone up to the number of zones specified in the configuration
+                        amp.getZoneIds().stream().limit(numZones).forEach((streamZoneId) -> {
+                            try {
+                                connector.queryZone(streamZoneId);
+                            } catch (MonopriceAudioException e) {
+                                logger.debug("Polling error: {}", e.getMessage());
```

Comment:
Consider adding `streamZoneId` in case the exception wouldn't contain this context.

Reply:
added!